src/builder-source-archive.c

```diff
@@ -638,18 +638,42 @@ static gboolean
 init_git (GFile   *dir,
           GError **error)
 {
-  char *basename;
+  char *basename, *user, *email;
+  gboolean ret;
 
   basename = g_file_get_basename (dir);
   if (!git (dir, NULL, error, "init", NULL) ||
-      !git (dir, NULL, error, "add", "--ignore-errors", ".", NULL) ||
-      !git (dir, NULL, error, "commit", "-m", basename, NULL))
+      !git (dir, NULL, error, "add", "--ignore-errors", ".", NULL))
     {
       g_free (basename);
       return FALSE;
     }
 
-  return TRUE;
+  /* Have an email for author and committer */
+  if (!git (dir, &email, NULL, "config", "--get", "user.email", NULL) ||
+      email == NULL)
+    email = g_strdup ("flatpak-builder-commit@flatpak.org");
+  else
+    email = trim_linefeed (email);
+  g_setenv ("GIT_AUTHOR_EMAIL", email, FALSE);
+  g_setenv ("GIT_COMMITTER_EMAIL", email, FALSE);
+  g_free (email);
```

Comment:
Overriding the environment globally like this feels a bit weird, as it will affect all processes that f-b later spawns.
Either we should pass the env vars specifically only to the child, or we should do this entire dance up-front in main() so that it affect all git calls.